axes/tests/test_handlers.py

```diff
@@ -199,6 +199,26 @@ def test_user_whitelisted(self, is_whitelisted):
         self.assertFalse(AxesProxyHandler().is_locked(self.request, self.credentials))
         self.assertEqual(1, is_whitelisted.call_count)
 
+    def test_user_login_failed_multiple_username(self):
+        settings_list = [
+            {},
+            {"AXES_ONLY_USER_FAILURES": True},
+            {"AXES_LOCK_OUT_BY_COMBINATION_USER_AND_IP": True},
+            {"AXES_USE_USER_AGENT": True},
+        ]
+        for setting_dict in settings_list:
+            with self.settings(**setting_dict):
+                self.login(username="admin")
+                self.assertEqual(
+                    1, AccessAttempt.objects.filter(username="admin").count()
```

Comment:
Could we check the number of failed attempts here as well from the `failures_since_start` property like so?

```
self.login(...)
attempts = AccessAttempt.objects.filter(...)
self.assertEqual(1, attempts.count())
self.assertEqual(1, attempts.first().failures_since_start)

self.login(...)
attempts = AccessAttempt.objects.filter(...)
self.assertEqual(1, attempts.count())
self.assertEqual(2, attempts.first().failures_since_start)
```

Also, using [`subTest`](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests) would make the possible test case possible more explicit:

```
for setting_dict in settings_list:
    with self.settings(**setting_dict):
        with self.subTest(settings=setting_dict):
           # ...
```

Reply:
Done.